extensions/2.0/Vendor/EXT_mesh_primitive_edge_visibility/README.md

```diff
@@ -0,0 +1,150 @@
+<figcaption><em><b>Figure 4</b> Edges drawn as separate primitives</em></figcaption>
+</figure>
+
+The [CESIUM_primitive_outline](https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Vendor/CESIUM_primitive_outline) extension provides an additional index buffer describing each hard edge as a line segment (a pair of indices into the triangle mesh's list of vertices). The extension leaves the details of how to render the edges up to the engine, requiring only that they render without depth-fighting. This approach satisfies the use case for which it was intended - displaying the edges of boxy, low-resolution buildings - but suffers some limitations:
+- The surface geometry must be represented as indexed triangles.
+- It only supports hard edges, not silhouettes.
+- The representation of the edges as pairs of vertex indices can significantly increase the size of the glTF asset.
+- The edges cannot specify their own materials.
+
+<figure>
+<img src="./figures/hard-edges.png"/>
+<figcaption><em><b>Figure 5</b> Hard edges encoded using CESIUM_primitive_outline</em></figcaption>
+</figure>
+
+## glTF Schema Updates
+
+The `EXT_mesh_primitive_edge_visibility` extension is applied to a mesh primitive of topology type 4 (triangles), 5 (triangle strip), or 6 (triangle fan) that uses indexed or non-indexed geometry. It provides the information required for engines to produce a rendering of the primitive's edges. Engines are not required to *always* render the edges - they may, for example, permit the user to toggle edge display on and off. When rendering the edges, engines may use any technique that meets the requirements specified below.
+
+### Edge Visibility
+
+The visibility of a single edge is specified using 2 bits as one of the following visibility values:
+- 0: Hidden edge - the edge should never be drawn.
+- 1: Silhouette edge - the edge should be drawn only in silhouette (i.e., when separating a front-facing triangle from a back-facing triangle).
+- 2: Hard edge - the edge should always be drawn.
+- 3: Hard edge encoded in `primitives` - a representation of the edge is included in the extension's `primitives` property. The edge should always be drawn, either by drawing the `primitives` array or by drawing it like an edge with visibility value `2` (never both).
+
+The extension's `visibility` property specifies the index of an accessor of `SCALAR` type and component type 5121 (unsigned byte) that encodes as a bitfield the visibility of every edge of every triangle in the mesh. The ordering of triangles and vertices is as described by [Section 3.7.2.1](https://www.khronos.org/registry/glTF/specs/2.0/glTF-2.0.html#meshes-overview) of the glTF 2.0 specification. For each triangle `(v0, v1, v2)`, the bitfield encodes three visibility values for the edges `(v0:v1, v1:v2, v2:v0)` in that order. Therefore, the accessor's `count` **MUST** be `6 * N / 8` rounded up to the nearest whole number, where `N` is the number of triangles in the primitive.
```

Comment:
It's an edge case (no pun intended), but should the spec define the value of bits when there are extra/unused bits at the end of the bitstream?

Reply:
Added requirement that unused bits are zero-initialized.